core/shared/src/main/scala/fs2/text.scala

```diff
@@ -360,7 +383,17 @@ object text {
           chunk.foreach { string =>
             fillBuffers(stringBuilder, linesBuffer, string)
           }
-          Pull.output(Chunk.buffer(linesBuffer)) >> go(stream, stringBuilder, first = false)
+
+          maxLineLength match {
+            case Some((max, raiseThrowable)) if stringBuilder.length > max =>
+              Pull.raiseError[F](
+                new IllegalStateException(
```

Comment:
Can anyone recommend any other type of exception here instead of IllegalStateException?

Reply:
I don't think it's an `IllegalStateException`.  We didn't call it at an inopportune time: we called it with an input that didn't match the configuration.  I think `IllegalArgumentException` is the closest, but I might just give it its own type of `RuntimeException`.

Reply:
Yeah that seems right to me. I've created a new `LineTooLongException`